### DISCUSSION TITLE
Does Quart load Jinja templates asynchronously?

### DISCUSSION THREAD

**Discussion by jakubsvehla** [pallets/quart#397] [Q&A]
Hi! I have looked into the source code and it seems that Jinja templates are loaded from the filesystem synchronously and blocking the async loop (which would kind of defeat the purpose of having an async framework), is that right? If so, is there a way to load the templates asynchronously and not block the loop?

**Comment by wigging** [comment]
I haven't looked at the source code of Quart but rendering a template is typically done with `await` as shown below which suggests that they are loaded asynchronously.

```python
from quart import render_template

@app.route("/hello")
async def hello():
    return await render_template("index.html")
```
